temporal-sdk/src/main/java/io/temporal/internal/replay/ReplayWorkflowTaskHandler.java

```diff
@@ -247,12 +247,16 @@ private Result createCompletedWFTRequest(
       }
       completedRequest.setStickyAttributes(attributes);
     }
-    if (!result.getSdkFlags().isEmpty()) {
-      completedRequest =
-          completedRequest.setSdkMetadata(
-              WorkflowTaskCompletedMetadata.newBuilder()
-                  .addAllLangUsedFlags(result.getSdkFlags())
-                  .build());
+    List<Integer> sdkFlags = result.getSdkFlags();
+    String writeSdkName = result.getWriteSdkName();
+    String writeSdkVersion = result.getWriteSdkVersion();
+    if (!sdkFlags.isEmpty() || writeSdkName != null || writeSdkVersion != null) {
+      completedRequest.setSdkMetadata(
+          WorkflowTaskCompletedMetadata.newBuilder()
+              .addAllLangUsedFlags(sdkFlags)
+              .setSdkName(writeSdkName)
+              .setSdkVersion(writeSdkVersion)
```

Comment:
I am not sure proto builder setters are null safe

Reply:
God they aren't. That's absolutely silly.

Reply:
:+1: May be showing a test coverage hole too. Alongside Quinn's suggestion of integration test, can we make sure there is some kind of test that would have failed if this code had not changed?